### PR TITLE
Adds prefix on `ssh_authenticator` call from `within`

### LINKED ISSUE #7: [Bug]: `ssh_authenticator` needs double colon prefix when used in `within`
### What happened?

`ssh_authenticator` needs double colon prefix in call when used in `within`

### sessionInfo()

_No response_

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by averissimo**
# Pull Request

Fixes #7

### How to test

- Have access to a server with ssh
- Have a `csv` file in that server
- Run code below adapting `PATH1`

```r
data <- teal.connectors.ssh::ssh_connector(
  paths = list(
    "PATH1" = "/path/to/file.csv"
  ),
  host = "127.0.0.1"
)

teal::init(
  data = data,
  modules = teal::modules(
    teal::example_module(label = "A"),
    teal::example_module(label = "B")
  )
)
```

### CODE REVIEW

**Comment by m7pr** [comment]
@averissimo do you recommend seting up any ssh server locally?
I see R CMD Check issues in your PR : /

```r
* checking dependencies in R code ... WARNING
'::' or ':::' import not declared from: ‘teal.connector.ssh’
```

File: R/ssh_connector.R [hunk 1]
```diff
@@ -142,7 +142,7 @@ ssh_connector <- function(data = teal.data::teal_data(),
           within(
             data,
             {
-              ssh_args <- ssh_authenticator(host = host)
+              ssh_args <- teal.connector.ssh::ssh_authenticator(host = host)
```
Comment (m7pr) [inline R/ssh_connector.R:145]: I think this triggers R CMD CHECK Warning 
https://github.com/insightsengineering/teal.connectors.ssh/actions/runs/10215201723/job/28264147809?pr=8#step:41:108
```r
* checking dependencies in R code ... WARNING
'::' or ':::' import not declared from: ‘teal.connector.ssh’
```

Comment (m7pr) [reply R/ssh_connector.R:145]: should we attach `teal.connector.ssh` (with `library`) in `data = teal_data()` to get rid of this warning?

Comment (averissimo) [reply R/ssh_connector.R]: That was a typo I made when committing :facepalm: sorry.

I got complacent as I expected a URL failing and didn't notice the R CMD check failing.

It's now corrected and I've also updated all the URLs (previously we were expecting the release, so some were pointing to `latest-tag`).

All green now

**Comment by averissimo** [comment]
Yes, I recommend using a local ssh server, but you can use one that I setup for testing:

```r
library(teal)
x <- teal.connectors.ssh::ssh_connector(
  host = "xxx.xxx.xxx.xxx",
  paths = list(ADSL = "/valid.csv", ADTTE = "/yada.csv")
)

init(data = x, modules = example_module()) |> 
  shiny::runApp()
```

I'll send the host, user and password via private message

**Comment by m7pr** [comment]
Hey, used data from `teal.data`

```r
library(teal.data)

write.csv(teal.data::rADSL, 'valid.csv')
write.csv(teal.data::rADTTE, 'yada.csv')

```

and successfully loaded csv files

image

**Review by m7pr — APPROVED** [review]

**Pull request merged**